### PR TITLE
Avoid creating multiple NoneShardSpec objects

### PR DESCRIPTION
**Pull request by nishantmonu51**

### CODE REVIEW

File: api/src/main/java/io/druid/timeline/partition/NoneShardSpec.java [hunk 1]
```diff
@@ -32,6 +33,11 @@
  */
 public class NoneShardSpec implements ShardSpec
 {
+  public final static NoneShardSpec INSTANCE = new NoneShardSpec();
```
Comment (gianm) [inline api/src/main/java/io/druid/timeline/partition/NoneShardSpec.java]: minor: consider having this private and using `instance()` everywhere

**Comment by gianm** [comment]
👍

File: api/src/main/java/io/druid/timeline/partition/NoneShardSpec.java (hunk 1 above)
Comment (nishantmonu51) [inline api/src/main/java/io/druid/timeline/partition/NoneShardSpec.java]: done.

File: api/src/main/java/io/druid/timeline/partition/NoneShardSpec.java [hunk 2]
```diff
@@ -32,6 +33,11 @@
  */
 public class NoneShardSpec implements ShardSpec
 {
+  private final static NoneShardSpec INSTANCE = new NoneShardSpec();
```
Comment (xvrl) [inline api/src/main/java/io/druid/timeline/partition/NoneShardSpec.java]: we could even make the constructor private to make sure instance is always used.

Comment (nishantmonu51) [inline api/src/main/java/io/druid/timeline/partition/NoneShardSpec.java]: @xvrl : I thought about it earlier but was left it public in order to not break any custom extensions using it with a minor version update in druid. If you feel strongly about it, let me know, i can make the change.

Comment (xvrl) [inline api/src/main/java/io/druid/timeline/partition/NoneShardSpec.java]: @nishantmonu51 maybe we can mark the constructor as deprecated then, so we can remove it in the future.

Comment (nishantmonu51) [inline api/src/main/java/io/druid/timeline/partition/NoneShardSpec.java]: done.

**Pull request merged**